extensions/sql-database-projects/src/controllers/projectController.ts

```diff
@@ -233,5 +236,191 @@ export class ProjectsController {
+			await project.addToProject(fileFolderList);
+
+			//Refresh project to show the added files
+			this.refreshProjectsTree();
+		}
+		catch (err) {
+			this.apiWrapper.showErrorMessage(utils.getErrorMessage(err));
+		}
+	}
+
+	private async getProjectName(dbName: string): Promise<string | undefined> {
+		let projName = await this.apiWrapper.showInputBox({
+			prompt: constants.newDatabaseProjectName,
+			value: `DatabaseProject${dbName}`
+		});
+
+		return projName;
+	}
+
+	private async getExtractTarget(): Promise<ExtractTarget> {
+		let extractTarget: ExtractTarget;
+
+		// eslint-disable-next-line code-no-unexternalized-strings
+		let extractTargetOptions: string[] = Object.keys(ExtractTarget).filter(k => typeof ExtractTarget[k as any] === "number");
+
+		let extractTargetInput = await this.apiWrapper.showQuickPickString(extractTargetOptions.slice(1), {
+			canPickMany: false,
+			placeHolder: constants.extractTargetInput
+		});
+
+		if (!extractTargetInput) {
+			// TODO: Default value of SchemaObjectType to be used or cancel out?
+			this.apiWrapper.showErrorMessage(constants.extractTargetDefault);
+			extractTarget = ExtractTarget.SchemaObjectType;
+		} else {
+			extractTarget = ExtractTarget[extractTargetInput as keyof typeof ExtractTarget];
+		}
+
+		return extractTarget;
+	}
+
+	private async getFolderLocation(extractTarget: ExtractTarget): Promise<Uri | undefined> {
+		let selectionResult;
+		let projUri;
+
+		if (extractTarget !== 1) {
+			selectionResult = await this.apiWrapper.showOpenDialog({
+				canSelectFiles: false,
+				canSelectFolders: true,
+				canSelectMany: false,
+				openLabel: constants.selectFileFolder,
+				defaultUri: this.apiWrapper.workspaceFolders() ? (this.apiWrapper.workspaceFolders() as WorkspaceFolder[])[0].uri : undefined
+			});
+			if (selectionResult) {
+				projUri = (selectionResult as Uri[])[0];
+			}
+		} else {
+			// Get filename
+			selectionResult = await this.apiWrapper.showSaveDialog(
+				{
+					defaultUri: this.apiWrapper.workspaceFolders() ? (this.apiWrapper.workspaceFolders() as WorkspaceFolder[])[0].uri : undefined,
+					saveLabel: constants.selectFileFolder,
+					filters: {
+						'Text files': ['sql'],
+						'All files': ['*']
+					}
+				}
+			);
+			if (selectionResult) {
+				projUri = selectionResult as unknown as Uri;
+			}
+		}
+
+		return projUri;
+	}
+
+	private async importApiCall(model: ImportDataModel): Promise<void> {
+		let ext = extensions.getExtension(mssql.extension.name);
+
+		if (ext === undefined) {
+			this.apiWrapper.showErrorMessage('VSCode extension undefined');
+			return;
+		}
+
+		const service = (ext.exports as mssql.IExtension).dacFx;
+		const ownerUri = await this.apiWrapper.getUriForConnection(model.serverId);
+
+		await service.importDatabaseProject(model.database, model.filePath, model.projName, model.version, ownerUri, model.extractTarget, TaskExecutionMode.execute);
+	}
+
+	private async generateList(absolutePath: string): Promise<string[]> {
```

Comment:
I agree to this. I will mostly review the DacFx code to get a list of files instead of reading it here.
For now, as we discussed- added the functionality to take as input only empty folders for the location of file- so that reading the folder and files goes as expected.